ccdproc/bitmask.py

```diff
@@ -0,0 +1,473 @@
+        >>> bm  # doctest: +SKIP
+        Bitmask (
+        array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        name_map={'hot': 1, 'low response': 3, 'bad': 0, 'artifact': 2}
+        )
+
+    Providing the names may be of help when creating a ``mask``:
+
+        >>> bm.to_mask(bitname=['low response'])
+        array([False, False, False, False, False, False, False, False,  True,  True], dtype=bool)
+    """
+    def __init__(self, bitmask, name_map=None):
+        self.data = bitmask
+        self.name_map = name_map
+
+    def __repr__(self):
+        return ('{self.__class__.__name__} (\n{self.data!r}'
+                '\nname_map={self.name_map}\n)'.format(self=self))
+
+    @property
+    def data(self):
+        return self._data
+
+    @data.setter
+    def data(self, val):
+        # Check if it looks like a "bitmask", meaning that it should be
+        # a `numpy.ndarray` and of integer data type.
+        val = np.asarray(val)
+        if val.dtype.kind not in _INTEGER_DTYPE_KINDS:
+            raise TypeError('"bitmask" must be of integer data type.')
+        self._data = val
+
+    @property
+    def name_map(self):
+        return self._name_map
+
+    @name_map.setter
+    def name_map(self, val):
+        if val is None:
+            val = {}
+        elif isinstance(val, (list, tuple)):
+            val = collections.OrderedDict(zip(val, itertools.count()))
+        elif isinstance(val, dict):
+            val = val
+        # Do the slow checks (collections) after the more common cases
+        elif isinstance(val, collections.Mapping):
+            val = val
+        elif isinstance(val, collections.Sequence):
+            val = collections.OrderedDict(zip(val, itertools.count()))
+        else:
+            raise TypeError('"name_map" must be a list/tuple/dict or another '
+                            'type of sequence/mapping, not a {}.'
+                            .format(type(val)))
+        self._name_map = val
+
+    @classmethod
+    def from_boolarray(cls, boolarray):
+        """Convert a boolean array representation of a bitmask to an integer
+        bitmask.
+
+        Parameters
+        ----------
+        boolarray : `numpy.ndarray` of boolean dtype with at least 2 dimensions
+            The boolean array to convert to a ``bitmask``. The function assumes
+            that the first dimension of the ``boolarray`` represents the
+            different flags.
+
+        Returns
+        -------
+        bitmask : `numpy.ndarray` of integer dtype
+            The bitmask representation of the ``boolarray``. It will have the
+            same shape as ``boolarray.shape[1:]``.
+
+        See also
+        --------
+        to_boolarray
+        ccdproc.boolarray_to_mask
+        """
+        boolarray = _prep_boolarray(boolarray)
+        # Number of layers (first dimension)
+        num = boolarray.shape[0]
+        # Choose the minimal dtype to represent the bitmask
+        dtype = _LAYERS_TO_DTYPE[num]
```

Comment:
Also, a note for the future: you may consider returning multi-layer arrays of flags... maybe...